generic/README.md

```diff
@@ -0,0 +1,157 @@
+This implementation overcomes this problem by extracting the language specific syntactic transformations to tree-sitter query API based rewrite rules, and applying them to the input program as chains of rules.
+
+<!-- ## Overview 
+
+At a higher level, Piranha updates the feature flag API usages related to the stale flag and then cleans up the resulting dead code. 
+Cleaning up the dead code involves edits like deleting the consequent of a if statement, or simplifying a boolean expression, or inlining a variable.
+
+We decompose the refactoring performed by Piranha into a chain (like `flat_map`/`chain`) of destructuve rewrites to the program.
+For instance, `if (a || exp.isTreated(STALE_FLAG)) { return a ; } return b;` would be replaced with `if (a || true) { return a ; } return b;` and then it with `if (true) { return a ; } return b;` and then with `return a ;`. -->
+
+## Usage 
+Piranha can be configured to recognize different flag APIs and flag behaviours by specifying a `input_rules.toml` file (and optionally a `input_edges.toml`) and the appropriate command line options. Moreover, Piranha can be configured to operate upon a new language by specifying a `rules.toml`, `edges.toml` and `scope_generators.toml`.
+```
+piranha-tree-sitter 0.1.0
+
+USAGE:
+    piranha-tree-sitter [OPTIONS] --path-to-codebase <PATH_TO_CODEBASE> --language <LANGUAGE> --flag-name <FLAG_NAME> --flag-namespace <FLAG_NAMESPACE> --path-to-configuration <PATH_TO_CONFIGURATION>
+
+OPTIONS:
+    -c, --path-to-configuration <PATH_TO_CONFIGURATION>
+            Folder containing the required configuration files
+    -f, --flag-name <FLAG_NAME>
+            Name of the stale flag
+    -h, --help
+            Print help information
+    -l, --language <LANGUAGE>
+    -n, --flag-namespace <FLAG_NAMESPACE>
+    -p, --path-to-codebase <PATH_TO_CODEBASE>
+            Input source code folder
+    -t, --flag-value
+            is treated?
+    -V, --version
+            Print version information
+```
+
+## Configuring Piranha  
+### Adding a flag API and specifying how it will be updated
+Consider the following definition of a rule in the `input_rules.toml` file. 
+```
+[[rules]]
+name = "Enum Based, toggle enabled"
+query = """((
+	(method_invocation 
+    	name : (_) @n1
+        arguments: ((argument_list 
+        				([
+                          (field_access field: (_)@f)
+                          (_) @f
+                         ])) )
+        	
+    ) @mi
+)
+(#eq? @n1 "isToggleEnabled")
+(#eq? @f "@stale_flag_name")
+)"""
+replace_node = "mi"
+replace = "@treated"
+groups = [ "Replace with boolean"]
+holes = ["treated", "stale_flag_name"]
+```
+This specifies a rule that matches against expressions like `exp.isToggleEnabled(SOME_FLAG_NAME)` and replaces it with `true` or `false`. 
```

Comment:
This will require a bit more explanation for folks unfamiliar with TreeSitter. Can we expand this discussion here?

Reply:
Addressed.